### PR TITLE
keptn 0.7.3 (new formula)

### PR DESCRIPTION
**Pull request by solidnerd**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/keptn.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  homepage "https://keptn.sh"
+  url "https://github.com/keptn/keptn/archive/0.7.3.tar.gz"
+  sha256 "7c5ab4bce7f8c75371a6130ae7929a9c7a88d05f29b96604b3849a93d2177228"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    Dir.chdir("cli")
+    system "go", "build", "-ldflags", "-s -w -X main.Version=#{version} -X main.KubeServerVersionConstraints=\"\"", "-trimpath", "-o", bin/"keptn"
```
Comment (christian-kreuzberger-dtx) [inline Formula/keptn.rb:12]: Any clue where `#{version}` is coming from? Or should this be fixed to 0.7.3?

Comment (solidnerd) [reply Formula/keptn.rb:12]: The version is autodetected from the URL and/or tag so only needs to be declared if it cannot be autodetected correctly.  

For more info look at:

https://rubydoc.brew.sh/Formula.html#stable-class_method

Comment (christian-kreuzberger-dtx) [reply Formula/keptn.rb:12]: Thanks for explaination :)

File: Formula/keptn.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  test do
+    run_output = shell_output("#{bin}/keptn version 2>&1")
+    assert_match "\nKeptn CLI version:", run_output
+
+    version_output = shell_output("#{bin}/keptn version 2>&1")
+    assert_match version.to_s, version_output
```
Comment (chenrui333) [inline Formula/keptn.rb:20]: we need some functional test than version test, you can x-check with the other go formulae.

File: Formula/keptn.rb (hunk 1 above)
Comment (chenrui333) [inline Formula/keptn.rb:12]:
```suggestion
    system "go", "build", *std_go_args, "-ldflags", "-s -w -X main.Version=#{version} -X main.KubeServerVersionConstraints=\"\"", "./cli"
```

File: Formula/keptn.rb (hunk 2 above)
Comment (solidnerd) [reply Formula/keptn.rb:19]: @chenrui333 Done. If something missing let me know :neckbeard:

**Comment by chenrui333** [comment]
let's also do arm build in this PR :)

**Comment by chenrui333** [comment]
test failed for arm
```
  Test::Unit::AssertionFailedError: </CLI\ is\ not\ authenticated\ against\ any\ Keptn\ cluster\./> was expected to be =~
<"\e[31mERRO\e[0m[0000] failure getting variant                       \e[31merror\e[0m=\"getCPUInfo for OS darwin: not implemented\"">.
```

**Comment by solidnerd** [comment]
@chenrui333  It seems that go support with version 1.16 darwin/arm64

**Comment by christian-kreuzberger-dtx** [comment]
Regarding darwin/arm support, we have https://github.com/keptn/keptn/issues/2297 

If you have any considerations or things that need to be done for ARM to be properly supported, please let us know. We're running with Go 1.13 atm.

**Review by chenrui333 — DISMISSED** [review]
lgtm

**Comment by chenrui333** [comment]
Thanks @solidnerd @christian-kreuzberger-dtx, looks good from my end.

**Comment by solidnerd** [comment]
I'm okay with a merge right now. It will solved in February with Go 1.16. Where we can bump the revision, or we got keptn 0.8 😁

**Comment by chenrui333** [comment]
Sounds good.

**Pull request closed without merging**